L1Trigger/L1THGCal/src/be_algorithms/HGCalShowerShape.cc

```diff
@@ -294,18 +300,27 @@ float HGCalShowerShape::sigmaPhiPhiMax(const l1t::HGCalMulticluster& c3d) const
 
 float HGCalShowerShape::eMax(const l1t::HGCalMulticluster& c3d) const {
   
-   const edm::PtrVector<l1t::HGCalCluster>& clustersPtrs = c3d.constituents();
+  std::unordered_map<int, float> layer_energy;
 
-   float EMax=0;   
+  const edm::PtrVector<l1t::HGCalCluster>& clustersPtrs = c3d.constituents();
+
+  for(const auto& clu : clustersPtrs){
+    
+    int layer = HGC_layer(clu->subdetId(),clu->layer());        
+    layer_energy[layer] += clu->energy();
 
-   for(const auto& clu : clustersPtrs){
+  }
+  
+  float EMax=0;   
+  
+  for(const auto& layer : layer_energy){
 
-     if(clu->energy()>EMax)
-       EMax = clu->energy();
+     if(layer.second>EMax)
+       EMax = layer.second;
```

Comment:
This is bug-prone. Better to either write the statement in a single line:
```cpp
if(layer.second>EMax) EMax = layer.second;
```
or to clearly delimit the statement with curly braces:
```cpp
if(layer.second>EMax) {
  EMax = layer.second;
}
```